### PR TITLE
Address additional Web Extension API feedback for Swift.

### PR DESCRIPTION
**Pull request by xeenon**
#### 9be0cc1962b926bda0d487fbdf23996dd0564aa9
<pre>
Address additional Web Extension API feedback for Swift.
<a href="https://webkit.org/b/279638">https://webkit.org/b/279638</a>
<a href="https://rdar.apple.com/problem/135959192">rdar://problem/135959192</a>

Reviewed by Jeff Miller.

Fixed the name for `WKWebExtensionContext.PermissionStatus`, which was missed in previous a rename.

Changed `didSelectTabs:` and `didDeselectTabs:` to take arrays instead of sets for better API ergonomics.

Removed all `NS_SWIFT_UNAVAILABLE` uses, so the optional parameter version is available to Swift without
the need to overlay many more methods.

Added some Swift overlays for `NS_REFINED_FOR_SWIFT` methods to give default values to optional parameters.

Dropped the remaining `conformsToProtocol:` checks in place of simple `nil` checks, as it is bad for
performance and not required as all our protocol methods are optional.

* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.mm:
(-[WKWebExtensionContext hasPermission:inTab:]):
(-[WKWebExtensionContext hasAccessToURL:inTab:]):
(-[WKWebExtensionContext permissionStatusForPermission:inTab:]):
(-[WKWebExtensionContext permissionStatusForURL:inTab:]):
(-[WKWebExtensionContext permissionStatusForMatchPattern:inTab:]):
(-[WKWebExtensionContext actionForTab:]):
(-[WKWebExtensionContext performActionForTab:]):
(-[WKWebExtensionContext menuItemsForTab:]):
(-[WKWebExtensionContext userGesturePerformedInTab:]):
(-[WKWebExtensionContext hasActiveUserGestureInTab:]):
(-[WKWebExtensionContext clearUserGestureInTab:]):
(-[WKWebExtensionContext didOpenWindow:]):
(-[WKWebExtensionContext didCloseWindow:]):
(-[WKWebExtensionContext didFocusWindow:]):
(-[WKWebExtensionContext didOpenTab:]):
(-[WKWebExtensionContext didCloseTab:windowIsClosing:]):
(-[WKWebExtensionContext didActivateTab:previousActiveTab:]):
(-[WKWebExtensionContext didSelectTabs:]):
(-[WKWebExtensionContext didDeselectTabs:]):
(-[WKWebExtensionContext didMoveTab:fromIndex:inWindow:]):
(-[WKWebExtensionContext didReplaceTab:withTab:]):
(-[WKWebExtensionContext didChangeTabProperties:forTab:]):
(-[WKWebExtensionContext sidebarForTab:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.mm:
(-[WKWebExtensionController didOpenWindow:]):
(-[WKWebExtensionController didCloseWindow:]):
(-[WKWebExtensionController didFocusWindow:]):
(-[WKWebExtensionController didOpenTab:]):
(-[WKWebExtensionController didCloseTab:windowIsClosing:]):
(-[WKWebExtensionController didActivateTab:previousActiveTab:]):
(-[WKWebExtensionController didSelectTabs:]):
(-[WKWebExtensionController didDeselectTabs:]):
(-[WKWebExtensionController didMoveTab:fromIndex:inWindow:]):
(-[WKWebExtensionController didReplaceTab:withTab:]):
(-[WKWebExtensionController didChangeTabProperties:forTab:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
* Source/WebKit/UIProcess/API/Cocoa/WebKitSwiftOverlay.swift:
(WKWebExtensionController.didCloseTab(_:windowIsClosing:)):
(WKWebExtensionController.didActivateTab(_:previousActiveTab:)):
(WKWebExtensionController.didMoveTab(_:from:in:)):
(WKWebExtensionContext.didCloseTab(_:windowIsClosing:)):
(WKWebExtensionContext.didActivateTab(_:previousActiveTab:)):
(WKWebExtensionContext.didMoveTab(_:from:in:)):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
(-[TestWebExtensionTab setSelected:forWebExtensionContext:completionHandler:]):

Canonical link: <a href="https://commits.webkit.org/283658@main">https://commits.webkit.org/283658@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b69e437a0a80cecb880d042c23fee095aa75c57c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66921 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46296 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19543 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70956 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18054 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69039 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54095 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17835 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/53671 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12152 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69988 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42591 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57907 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/34214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39262 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15299 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16408 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/61182 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15640 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72657 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10878 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14985 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10910 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57964 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/61250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8937 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2557 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10159 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/42103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43180 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44363 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42923 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h [hunk 1]
```diff
@@ -25,6 +25,8 @@
 
 #import <WebKit/WKWebExtensionContextPrivate.h>
 
+#define HAVE_WK_WEB_EXTENSION_CONTEXT_ARRAY_BASED_DID_SELECT_TABS 1
```
Comment (jeffmapple) [inline Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:28]: Was this change intentional?

**Review by jeffmapple — APPROVED** [review]

File: Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h (hunk 1 above)
Comment (jeffmapple) [reply Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:28]: Oh, I see this is used for staging internally.

Comment (xeenon) [reply Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:28]: Yep
